src/lib/pages/backoffice/Providers/ProviderEditor/ProviderEditor.js

```diff
@@ -0,0 +1,128 @@
+import { providerApi } from '@api/providers';
+import { withCancel } from '@api/utils';
+import { Error } from '@components/Error';
+import { Loader } from '@components/Loader';
+import { RJSForm } from '@forms/rjsf';
+import { goTo } from '@history';
+import { ProviderRoutes } from '@routes/urls';
+import _get from 'lodash/get';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { schema } from './schema';
+import { uiSchema } from './uiSchema';
+
+export class ProviderEditor extends Component {
+  constructor(props) {
+    super(props);
+
+    const isEditing = !!props.match.params.providerPid;
+    this.state = {
+      data: {},
+      isLoading: isEditing,
+      error: {},
+    };
+  }
+
+  componentDidMount() {
+    const { match } = this.props;
+    if (match.params.providerPid) {
+      this.fetchProvider(match.params.providerPid);
+    }
+  }
+
+  componentWillUnmount() {
+    this.cancellableFetchProvider && this.cancellableFetchProvider.cancel();
+  }
+
+  fetchProvider = async (providerPid) => {
+    this.cancellableFetchProvider = withCancel(providerApi.get(providerPid));
+    try {
+      const response = await this.cancellableFetchProvider.promise;
+      this.setState({ data: response.data, isLoading: false, error: {} });
+    } catch (error) {
+      if (error !== 'UNMOUNTED') {
+        this.setState({ isLoading: false, error: error });
+      }
+    }
+  };
+
+  submitAction = async (formData) => {
+    const {
+      match: {
+        params: { providerPid },
+      },
+    } = this.props;
+
+    const isEditing = !!providerPid;
+    return isEditing
+      ? await providerApi.update(providerPid, formData)
+      : await providerApi.create(formData);
+  };
+
+  successCallback = (response) => {
+    goTo(
+      ProviderRoutes.providerDetailsFor(_get(response, 'data.metadata.pid'))
```

Comment:
Here I copied/pasted everywhere, but the `pid` must be there, maybe you can fix it everywhere with a find/replace:
```suggestion
      ProviderRoutes.providerDetailsFor(response.data.metadata.pid)
```